src/cdk-new/create-rest-api.ts

```diff
@@ -0,0 +1,42 @@
+import type {RestApiBase, StageOptions} from 'aws-cdk-lib/aws-apigateway';
+import {
+  EndpointType,
+  RestApi,
+  SecurityPolicy,
+} from 'aws-cdk-lib/aws-apigateway';
+import type {ICertificate} from 'aws-cdk-lib/aws-certificatemanager';
+import type {Stack} from 'aws-cdk-lib/core';
+import {CfnOutput} from 'aws-cdk-lib/core';
+
+export interface RestApiInit {
+  readonly stack: Stack;
+  readonly certificate: ICertificate;
+  readonly restApiName: string;
+  readonly domainName: string;
+  readonly stageOptions: StageOptions;
+}
+
+export function createRestApi(init: RestApiInit): RestApiBase {
+  const {stack, certificate, restApiName, domainName, stageOptions} = init;
+
+  const restApi = new RestApi(stack, `RestApi`, {
+    endpointTypes: [EndpointType.REGIONAL],
+    restApiName,
+    domainName: {
+      endpointType: EndpointType.REGIONAL,
+      domainName,
+      certificate,
+      securityPolicy: SecurityPolicy.TLS_1_2,
+    },
+    disableExecuteApiEndpoint: true,
+    binaryMediaTypes: [`*/*`],
```

Comment:
`*/*` looks dangerous.

Reply:
.. and would likely break our application

Reply:
No this is documented and has something to do with the handling of binary and non-binary files. But this takes a little longer to explain. Gladly by teams ;)